src/login/logind-button.c

```
@@ -117,6 +117,40 @@ static int button_install_check_event_source(Button *b) {
         return sd_event_source_set_priority(b->check_event_source, SD_EVENT_PRIORITY_IDLE+1);
 }
 
+static int long_press_of_reboot_key_handler(sd_event_source *e, uint64_t usec, void *userdata) {
+        Manager *m = userdata;
+
+        assert(e);
+        assert(m);
+
+        m->reboot_key_long_press_event_source = sd_event_source_unref(m->reboot_key_long_press_event_source);
+
+        log_struct(LOG_INFO,
+                   LOG_MESSAGE("Reboot key long press detected."),
+                   "MESSAGE_ID=" SD_MESSAGE_REBOOT_KEY_LONGPRESS_STR);
+
+        manager_handle_action(m, INHIBIT_HANDLE_REBOOT_KEY, m->handle_reboot_key_long_press, m->reboot_key_ignore_inhibited, true);
+        return 0;
+}
+
+static void start_long_press_of_reboot_key(Manager *m) {
+        int r;
+
+        assert(m);
+
+        if (m->reboot_key_long_press_event_source)
+                return;
+
+        r = sd_event_add_time_relative(
+                        m->event,
+                        &m->reboot_key_long_press_event_source,
+                        CLOCK_MONOTONIC,
+                        m->long_press_duration, 0,
+                        long_press_of_reboot_key_handler, m);
+        if (r < 0)
+                log_error_errno(r, "Failed to add long press timer event");
```

since we ignore the error here this should be a warning only. also, it should say "ignoring" at the end, since we ignore it, and we should output the error string for the error here. i.e.

```c
if (r < 0)
        log_warning_errno(r, "Failed to add long press timer event, ignoring: %m");
```